specs/error-handling-guideline.md

```diff
@@ -0,0 +1,65 @@
+# Notation CLI Error Handling and Message Guideline
+
+This document aims to provide the guidelines for Notation contributors to improve existing error messages and error handling method as well as the new error output format. It will also provide recommendations and examples for Notation CLI contributors for how to write friendly and standard error messages, avoid generating inconsistent and ambiguous error messages.
+
+## General guiding principles
+
+A clear and actionable error message is very important when raising an error, so make sure your error message describes clearly what the error is and tells users what they need to do if possible.
+
+First and foremost, make the error messages descriptive and informative. Error messages are expected to be helpful to troubleshoot where the user has done something wrong and the program is guiding them in the right direction. A great error message is recommended to contain the following elements:
+
+- HTTP status code: optional, when the logs are generated from the server side, it's recommended to print the HTTP status code in the error description
+- Error description: describe what the error is
+- Suggestion: for those errors that have potential solution, print out the recommended solution. Versioned troubleshooting document link is nice to have
```

Comment:
Dont add document link to error message, websites will get restructured and link might become inaccessible

Reply:
We recently supported versioned links to the Notary Project documentation so the broken links will not happen. 
If there is a long information in the error message as a recommended action, how can `notation` effectively prompt users without giving a troubleshooting doc link?